Serializer/Hateoas/SonataRelationProviderResolver.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/*
+ * This file is part of the Sonata package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ */
+namespace Sonata\AdminBundle\Serializer\Hateoas;
+
+use Hateoas\Configuration\Provider\Resolver\RelationProviderResolverInterface;
+use Sonata\AdminBundle\Admin\Pool;
+use Hateoas\Configuration\RelationProvider;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Hateoas\Configuration\Relation;
+use Hateoas\Configuration\Route;
+
+/**
+ * @author Daniel Leech <daniel@dantleech.com>
+ */
+class SonataRelationProviderResolver implements RelationProviderResolverInterface
+{
+    /**
+     * @var Pool
+     */
+    private $pool;
+
+    /**
+     * @var UrlGeneratorInterface
+     */
+    private $urlGenerator;
+
+    /**
+     * @param Pool $pool
+     * @param UrlGeneratorInterface $urlGenerator
+     */
+    public function __construct(Pool $pool, UrlGeneratorInterface $urlGenerator)
+    {
+        $this->pool = $pool;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getRelationProvider(RelationProvider $relationProvider, $object)
+    {
+        if (false === $this->pool->hasAdminByClass(get_class($object))) {
+            return null;
+        }
+
+        return array($this, 'getRelations');
+    }
+
+    /**
+     * Return the sonata admin relations
+     *
+     * @return array
+     */
+    public function getRelations($object)
```

Comment:
The second link you gave is the same as the first .. 

If a class has more than one admin I think it would be good somehow to have something as follows:

````
{
    '_links': {
        '<admin1 name>_edit': '/path/to',
        '<admin1 name>_view': '/path/to',
        '<admin1 name>_delete': '/path/to',
        '<admin2 name>_edit': '/path/to',
        '<admin2 name>_view',: '/path/to'
        '<admin2 name>_delete': '/path/to'
    }
}
````

So there are two problems:

- How do I get all of the admin classes for a given class? Seems easy to add if I'm not missing something.
- And I was going to say how do I get the short name of the admin, but I guess I can just use the generated route names as the keys about